examples/java/core/nothing_default/src/main/java/com/microsoft/bond/examples/NothingDefault.java

```diff
@@ -0,0 +1,87 @@
+package com.microsoft.bond.examples;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.LinkedList;
+
+import com.microsoft.bond.BondSerializable;
+import com.microsoft.bond.Deserializer;
+import com.microsoft.bond.Serializer;
+import com.microsoft.bond.Something;
+import com.microsoft.bond.StructBondType;
+
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.CompactBinaryWriter;
+
+import com.microsoft.bond.examples.nothingdefault.Struct_v1;
+import com.microsoft.bond.examples.nothingdefault.Struct_v2;
+
+public class NothingDefault {
+
+    public static void main(final String[] args) throws IOException {
+
+        final Struct_v1 v1 = new Struct_v1();
+
+        boolean sawException = false;
+
+        // Struct_v1 has a required field foo which by default is set to 'nothing'.
+        // If we try to serialize object v1 w/o initializing the field to some value
+        // Bond will throw an exception.
+        try {
+            System.out.println("Serializing v1...");
+            serialize(v1);
```

Comment:
I expected serialization of Struct_v1 in its current state to throw, but it doesn't. Looking at the code, it seems like we're never checking whether the field is required before we skip serializing it because it is null.

The generated code calls `StructBondType.SomethingInt32Field()`:

```
@Override
protected final void serializeStructFields(com.microsoft.bond.BondType.SerializationContext context, Struct_v1 value) throws java.io.IOException {
    this.foo.serialize(context, value.foo);
    this.baz.serialize(context, value.baz);
}
```

From [`StructBondType.SomethingInt32Field()`](https://github.com/Microsoft/bond/blob/jvm/java/core/src/main/java/com/microsoft/bond/StructBondType.java#L1639-L1642):

```
public final void serialize(
    SerializationContext context, SomethingInteger value) throws IOException {
    Int32BondType.serializePrimitiveSomethingField(context, value, this);
}
```

From [`Int32BondType.serializePrimitiveSomethingField()`](https://github.com/Microsoft/bond/blob/jvm/java/core/src/main/java/com/microsoft/bond/Int32BondType.java#L167-L174):

```
protected static void serializePrimitiveSomethingField(
        SerializationContext context,
        SomethingInteger value,
        StructBondType.StructField<Integer> field) throws IOException {
    if (value != null) {
        serializePrimitiveField(context, value.value, field);
    }
}
```